Resources/public/js/views/ez-testonearticlesview.js

```diff
@@ -0,0 +1,195 @@
+                container.append(LOADER);
+            }
+        },
+
+        /**
+         * Event handler for article loaded
+         *
+         * @method _articleLoaded
+         * @param {Object} e event facade
+         * @protected
+         */
+        _articleLoaded: function (e) {
+            this.set('currentLocation', e.currentLocation);
+            this.get('container').removeClass(ARTICLES_CLASS_PREFIX + '--loading');
+            LOADER.detach();
+            this.set('articles', e.articles);
+        },
+
+        /**
+         * Event handler for selecting article link
+         *
+         * @method _selectArticle
+         * @param {Object} e event facade
+         * @protected
+         */
+        _selectArticle: function (e) {
+            e.preventDefault();
+            var selectedArticle = this.get('selectedArticle'),
+                articles = this.get('articles'),
+                locationId = e.target.getAttribute('data-location-id'),
+                history = this.get('history') || [],
+                currentLocation = this.get('currentLocation');
+
+            if (selectedArticle && selectedArticle.target) {
+                selectedArticle.target.removeClass('article__link--active');
+            }
+
+            if (e.target.hasClass('article__link--previous') && history.length > 0) {
+                selectedArticle = history.pop();
+                articles[locationId] = selectedArticle.article;
+                this.set('history', history);
+            } else if (articles[locationId].hasChildren) {
+                history.push({
+                    article: currentLocation
+                });
+                this.set('history', history);
+            }
+
+            if (!articles[locationId]) {
+                return;
+            }
+
+            selectedArticle = {
+                target: e.target,
+                article: articles[locationId]
+            };
+            e.target.addClass('article__link--active');
+            this.set('selectedArticle', selectedArticle);
+        },
+
+        /**
+         * Event handler for changing root location of article list
+         *
+         * @method _changeRootLocation
+         * @param {Object} e event facade
+         * @protected
+         */
+        _changeRootLocation: function (e) {
+            var article = e.newVal;
+
+            this.set('selectedArticle', {
+                locationId: article.locationId,
+                article: article
+            });
+        }
+    }, {
+        ATTR: {
+            /**
+             * Article list of current location
+             *
+             * @attribute articles
+             * @default {}
+             * @type {Object}
+             */
+            articles: {
+                value: {}
+            },
+
+            /**
+             * Currently selected article
+             *
+             * @attribute selectedArticle
+             * @default null
+             * @type {Object}
```

Comment:
Jeśli oczekujesz obiektu to nie możesz dawać nulla jako wartość domyślną. W takiej sytuacji lepiej mieć taki zapis: `selectedArticle: {},`